service/src/main/java/bio/terra/workspace/service/iam/SamService.java

```diff
@@ -130,6 +126,31 @@ public String getUserEmailFromSam(AuthenticatedUserRequest userRequest)
     }
   }
 
+  /**
+   * Fetch a user-assigned managed identity that was created for a user, with user credentials,
+   * directly from Sam.
+   */
+  public String getOrCreateUserManagedIdentity(
+      AuthenticatedUserRequest userRequest,
+      String subscriptionId,
+      String tenantId,
+      String managedResourceGroupId)
+      throws InterruptedException {
+    AzureApi azureApi = samAzureApi(userRequest.getRequiredToken());
+
+    GetOrCreatePetManagedIdentityRequest request =
+        new GetOrCreatePetManagedIdentityRequest()
+            .subscriptionId(subscriptionId)
+            .tenantId(tenantId)
+            .managedResourceGroupName(managedResourceGroupId);
+    try {
+      return SamRetry.retry(() -> azureApi.getPetManagedIdentity(request));
+    } catch (ApiException apiException) {
+      throw SamExceptionFactory.create(
```

Comment:
Should we log error?

Reply:
SamExceptionFactory is taking care of logging error. Following existing pattern here.